File: Software Development Module/Week3/NumbersApp.java
/*
NumbersApp.java
Billy Cussen 
08/07/2020
*/

//This is the App Class for Numbers.java. This is the executable class

import java.util.InputMismatchException;
import java.util.Scanner;

public class NumbersApp {
    
    public static void main(String[]args){

        //Variables
        int a = 0, b = 0, sum;
        boolean checkInputA, checkInputB;

        //Objects
        Scanner keyboard;
        Numbers myNumbers = new Numbers();

        //Input
        do{

            checkInputA = true;
            keyboard = new Scanner(System.in);

            try{
                System.out.print("Please enter number A: ");
                a = keyboard.nextInt();

            }catch(InputMismatchException e){
                System.out.println("This is not an Integer, please try again");
                checkInputA = false;
            }

        }while(!checkInputA);

        do{

            checkInputB = true;
            keyboard = new Scanner(System.in);

            try{
                System.out.print("Please enter number B: ");
                b = keyboard.nextInt();

            }catch(InputMismatchException e){
                System.out.println("This is not an Integer, please try again");
                checkInputB = false;
            }

        }while(!checkInputB);

        //Setters
        myNumbers.setA(a);
        myNumbers.setB(b);

        //Compute
        myNumbers.computeSum();

        //Output
        sum = myNumbers.getSum();

        System.out.println("The sum of "+a+" and "+b+" is: "+sum);

    }
}

File: Python/Software Development Module - Python/Week3/NumbersApp.py
""""
NumbersApp.py
Billy Cussen
25/09/2020
"""

import Numbers
from Numbers import myNumbers

inputCheck = True
repeat = "y"

while repeat == "y":

    #User Input - Number 1
    while inputCheck:

        try:
            one = int(input("Please enter Number 1: "))
        except:
            print("You need to enter an Integer")
        else:
            if one <= 0:
                print("Your input needs to be greater than 0")
            else:
                inputCheck = False

    #User Input - Number 2
    while inputCheck == False:

        try:
            two = int(input("Please enter Number 2: "))
        except:
            print("You need to enter an Integer")
        else:
            if two <= 0:
                print("Your input needs to be greater than 0")
            else:
                inputCheck = True

    #Process & Output
    numberApp = myNumbers(one,two)
    print("The Sum of these Numbers is: "+str(numberApp.getSum()))

    repeat = ""

    while repeat != "y" and repeat != "n":
        repeat = str(input("Please enter if you would like to retry or not (\"y\" or \"n\"): "))
        repeat = repeat.lower()
        if(repeat != "y" and repeat != "n"):
            print("You need to enter Y or N")

print("Thanks for using the App!")